File: switch.py
import os
import shutil
import sys

# Pobranie argumentów z wiersza poleceń
args = sys.argv[1:]

# Sprawdzenie, czy użytkownik podał poprawnie argumenty
if len(args) != 2:
    print('Usage: python switch.py <module you want to export> <module you want to import>')
    sys.exit(1)

# Pobranie nazwy folderu, do którego mają zostać przeniesione (wyeksportowane) zasoby
expArg = args[0]

# Pobranie nazwy folderu, z którego mają zostać wyciągnięte (zaimportowane) zasoby
impArg = args[1]

# Ścieżka do folderu ./data
exp_source_path = './data'

# Ścieżka do folderu źródłowego (./_modules/impArg)
imp_source_path = f'./_modules/{impArg}'

# Ścieżka do folderu docelowego (./_modules/expArg)
exp_destination_path = f'./_modules/{expArg}'

# Ścieżka do folderu docelowego (./data)
imp_destination_path = './data'

# Sprawdzenie, czy folder ./data istnieje
if not os.path.exists(exp_source_path):
    print('Folder ./data does not exist')
    sys.exit(1)

# Sprawdzenie, czy folder docelowy (./_modules/expArg) istnieje
if not os.path.exists(exp_destination_path):
    print(f'Folder ./_modules/{expArg} does not exist')
    sys.exit(1)

# Sprawdzenie, czy folder docelowy zawiera już podfolder 'data'
data_folder_path = os.path.join(exp_destination_path, 'data')
if os.path.exists(data_folder_path):
    print(f'The destination folder already contains a data subfolder')
    sys.exit(1)

# Sprawdzenie, czy folder ./_modules/<impArg> istnieje
if not os.path.exists(imp_source_path):
    print(f'Folder ./_modules/{impArg} does not exist')
    sys.exit(1)


#*############################### moving folders #################################

try:
    # Przeniesienie folderu ./data do folderu docelowego
    shutil.move(exp_source_path, os.path.join(exp_destination_path, 'data'))
    print('Folder ./data exported successfully')
except Exception as err:
    print('Error moving folder:', err)
    sys.exit(1)

try:
    # Przeniesienie folderu ./_modules/<impArg>/data do folderu docelowego
    shutil.move(os.path.join(imp_source_path, 'data'), imp_destination_path)
    print('Folder data from ./_modules imported successfully')
except Exception as err:
    print('Error moving folder:', err)

    sys.exit(1)

File: switch.js
const fs = require('fs')
const path = require('path')

// Pobranie argumentów z wiersza poleceń
const args = process.argv.slice(2)
console.log("🚀 ~ args:", args)

// Sprawdzenie, czy użytkownik podał argumenty
if (args.length !== 2) {
  console.error(
    'Usage: node switch <module you want to export> <module you want to import>'
  )
  process.exit(1)
}

// Pobranie nazwy folderu, do którego mają zostać przeniesione zasoby
const arg1 = args[0]

// Ścieżka do folderu ./data
const sourcePath = './data'

// Ścieżka do folderu docelowego (./_modules/arg1)
const destinationPath = `./_modules/${arg1}`

console.log("🚀 ~ destinationPath:", destinationPath)
// Sprawdzenie, czy folder ./data istnieje
if (!fs.existsSync(sourcePath)) {
  console.error('Folder ./data does not exist')
  process.exit(1)
}

// Sprawdzenie, czy folder docelowy (./_modules/arg1) istnieje
if (!fs.existsSync(destinationPath)) {
  console.error(`Folder ./_modules/${arg1} does not exist`)
  process.exit(1)
}

try {
  // Przeniesienie folderu ./data do folderu docelowego
  fs.renameSync(sourcePath, path.join(destinationPath, 'data'))
  console.log('Folder ./data moved successfully')
} catch (err) {
  console.error('Error moving folder:', err)
  process.exit(1)
}

//* importowanie nowego folderu

// Pobranie nazwy folderu, z którego mają zostać wyciągnięte zasoby
const arg2 = args[1]

// Ścieżka do folderu źródłowego (./_modules/arg2)
const sourcePath2 = `./_modules/${arg2}`

// Ścieżka do folderu docelowego (./data)
const destinationPath2 = './data'

// Sprawdzenie, czy folder ./_modules/<arg2> istnieje
if (!fs.existsSync(sourcePath2)) {
  console.error(`Folder ./_modules/${arg2} does not exist`)
  process.exit(1)
}

try {
  // Przeniesienie folderu ./_modules/<arg2>/data do folderu docelowego
  fs.renameSync(path.join(sourcePath2, 'data'), destinationPath2)
  console.log('Folder data from ./_modules moved successfully')
} catch (err) {
  console.error('Error moving folder:', err)
  process.exit(1)
}
